### PR TITLE
Add basic support for FTS

### PR DESCRIPTION
**Pull request by bugthunk**
* add support for specifying `BTree` or `GIN` indexes
* add `tsvector` type
* also, bump version because of the first point above

### CODE REVIEW

File: src/Database/PostgreSQL/PQTypes/Checks.hs [hunk 1]
```diff
@@ -842,10 +845,11 @@ sqlGetIndexes table = toSQLCommand . sqlSelect "pg_catalog.pg_class c" $ do
       , "SELECT name FROM coordinates WHERE k > 0"
       ]
 
-fetchTableIndex :: (String, Array1 String, Bool, Maybe String)
+fetchTableIndex :: (String, Array1 String, String, Bool, Maybe String)
                 -> (TableIndex, RawSQL ())
-fetchTableIndex (name, Array1 columns, unique, mconstraint) = (TableIndex {
+fetchTableIndex (name, Array1 columns, method, unique, mconstraint) = (TableIndex {
   idxColumns = map unsafeSQL columns
+, idxMethod = read $ map toLower method
```
Comment (bugthunk) [inline src/Database/PostgreSQL/PQTypes/Checks.hs]: `toLower` is perhaps an unnecessary precaution.

**Comment by bugthunk** [comment]
No tests were added for these features since I don't have the time right now - I've created a ticket with @jonathanjouty to rectify the situation.

File: hpqtypes-extras.cabal [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name:                hpqtypes-extras
-version:             1.3.1.1
+version:             1.4
```
Comment (23Skidoo) [inline hpqtypes-extras.cabal]: Please use a 4-component version number (i.e. `1.4.0.0`) for consistency.

File: src/Database/PostgreSQL/PQTypes/Model/Index.hs [hunk 3]
```diff
@@ -24,40 +27,74 @@ import qualified Data.Text.Encoding as T
...
+    BTree
+  | GIN
+  deriving (Eq, Ord)
+
+instance Show IndexMethod where
+    show BTree = "btree"
+    show GIN   = "gin"
+
+instance Read IndexMethod where
+    readsPrec _ "btree" = [(BTree,"")]
```
Comment (23Skidoo) [inline src/Database/PostgreSQL/PQTypes/Model/Index.hs]: Yeah, I'd make this case-insensitive. You can use view patterns: `readsPrec _ (map toLower -> "btree") = ...`.

File: hpqtypes-extras.cabal (hunk 2 above)
Comment (23Skidoo) [reply hpqtypes-extras.cabal]: Also please update the changelog.

Comment (bugthunk) [reply hpqtypes-extras.cabal]: gotcha

**Comment by bugthunk** [comment]
Fixed all of the above.

File: src/Database/PostgreSQL/PQTypes/Model/Index.hs [hunk 4]
```diff
@@ -89,7 +126,8 @@ sqlCreateIndex :: RawSQL () -> TableIndex -> RawSQL ()
 sqlCreateIndex tname idx@TableIndex{..} = mconcat [
     "CREATE "
   , if idxUnique then "UNIQUE " else ""
```
Comment (bugthunk) [inline src/Database/PostgreSQL/PQTypes/Model/Index.hs]: @23Skidoo Only `BTree` indexes can ever be unique, so I guess we really should ignore any method supplied or throw. For the future?

**Comment by bugthunk** [comment]
@23Skidoo don't merge just yet, OK?

File: src/Database/PostgreSQL/PQTypes/Model/Index.hs (hunk 4 above)
Comment (23Skidoo) [reply src/Database/PostgreSQL/PQTypes/Model/Index.hs]: Shouldn't be too hard to add a check, I think.

**Comment by 23Skidoo** [comment]
@bugthunk ack

File: src/Database/PostgreSQL/PQTypes/Model/Index.hs (hunk 4 above)
Comment (bugthunk) [reply src/Database/PostgreSQL/PQTypes/Model/Index.hs]: No, I just thought I was out of time but there's a weekend ahead so I'll probably do it then.

**Comment by bugthunk** [comment]
@23Skidoo @jonathanjouty nah, this is good to go. IMHO we shouldn't care about the `unique`/`gin` combination since the runtime error from PG is enough:
```
# create unique index idx_throwaway on someTable using gin (someColumn);
ERROR:  access method "gin" does not support unique indexes
```

**Comment by 23Skidoo** [comment]
Released version 1.4.0.0 on Hackage.

**Pull request merged**